### PR TITLE
share the dns resolver

### LINKED ISSUE #56: Do not create a new DNS resolver for each new connection
Currently a new DNS resolver is created for each new connection. Since each resolver gets its own task spawned on the executor there are a lot of unused tasks created. The resolver can be kept as a field of the connection factory. Furthermore the AsyncResolver implements clone which means that resolver clones can be passed in from the outside and therefore be shared

### PR DESCRIPTION
**Pull request by chridou**
closes #56

### CODE REVIEW

**Comment by chridou** [comment]
This is not pefect. If a new pool is created still a new dns resolver would be created...

**Comment by chridou** [comment]
:+1:

**Comment by chridou** [comment]
Actually the DNS resolver has an unbounded channel under the hood. This means the background tsk should stop once the last resolver is gone. But nevertheless one might still want to share/keep the resolver since its creation can theoretically fail.

File: src/config/builder.rs [hunk 1]
```diff
@@ -272,7 +272,21 @@ impl Builder {
 
     /// Build a new `RedisPool`
     pub fn finish_redis_rs(self) -> InitializationResult<RedisPool> {
-        self.finish(RedisRsFactory::new)
+        let (resolver, background_task) = trust_dns_resolver::AsyncResolver::from_system_conf()
```
Comment (panicbit) [inline src/config/builder.rs:275]: We can use [read_system_conf](https://docs.rs/trust-dns-resolver/0.18.0/trust_dns_resolver/system_conf/fn.read_system_conf.html) to read resolv.conf separately, allowing us to tweak the resolver options such as [cache_size](https://docs.rs/trust-dns-resolver/0.18.0/trust_dns_resolver/config/struct.ResolverOpts.html#structfield.cache_size)

**Pull request closed without merging**